src/algorithms/digi/PhotoMultiplierHitDigi.cc

```diff
@@ -142,8 +145,62 @@ eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(std::vector<const edm4hep::Si
                 raw_hits.push_back(hit);
             }
         }
-        return raw_hits;
 
+        if (m_cfg.noiseInjection) {
+          //build noise raw hits
+          std::unordered_map<uint64_t, std::vector<HitData>> hit_groups_noise;
+
+          const auto& readoutCoder = *m_detector->readout("DRICHHits").idSpec().decoder();
+          auto m_detRICH = m_detector->detector("DRICH");
+          auto systemID = m_detRICH.id();
+          auto m_posRICH = m_detRICH.placement().position();
+          auto nSectors = m_detector->constant<int>("DRICH_num_sectors");
+          auto num_px = m_detector->constant<int>("DRICH_num_px");
+
+          for (int isec = 0; isec < nSectors; isec++) {
+            std::string secName = "sec" + std::to_string(isec);
+
+            for(auto const& [de_name, detSensor] : m_detRICH.children()) {
+              if(de_name.find("sensor_de_"+secName)!=std::string::npos) {
+                // get sensors info
+                auto imodsec = detSensor.id();
+                uint64_t x_encoded = 0;
+
+                for (uint64_t x = 0; x < num_px; x++){
+                  x_encoded = ( x << readoutCoder["x"].offset() )  & readoutCoder["x"].mask();
+                  uint64_t y_encoded = 0;
+                  for (uint64_t y = 0; y < num_px; y++){
+                    y_encoded = ( y << readoutCoder["y"].offset() ) & readoutCoder["y"].mask();
+                    auto cellID = systemID + imodsec + x_encoded + y_encoded;
+                    auto it = hit_groups.find(cellID);
+                    if (Noise_Digits(m_cfg.noiseRate, m_cfg.timeWindow)){
+                      // cell time, signal amplitude
+                      double amp = m_cfg.speMean + m_rngNorm()*m_cfg.speError;
+                      double time = m_cfg.timeWindow*m_rngUni();
+                      dd4hep::Position pos_hit_global = m_cellid_converter->position(cellID);
+                      dd4hep::Position pos_hit = get_sensor_local_position( cellID, pos_hit_global );
```

Comment:
Hmm... now I see why you originally wanted to change `HitData::pos`. The only reason I want to keep `pos_hit = get_sensor_local_position(...)` for the actual signal hits is to check if the pixel gap cuts are working correctly; but I fear the `get_sensor_local_position` function is kind of slow. Here in this noise injection loop, you don't have a need to apply pixel gap cuts, so you could just leave this as empty (and when we want to check pixel gap cuts, we just disable noise injection). So to speed things up, I suggest:
```suggestion
                      dd4hep::Position pos_hit{0,0,0}; // unused
```